colcon_core/command.py

```diff
@@ -226,7 +226,14 @@ class CustomArgumentParser(argparse.ArgumentParser):
 
         def _parse_optional(self, arg_string):
             result = super()._parse_optional(arg_string)
-            if result == (None, arg_string, None):
+            # Up until https://github.com/python/cpython/pull/114180 ,
+            # _parse_optional() returned a 3-tuple when it couldn't classify
+            # the option.  As of that PR (which is in Python 3.13, and
+            # backported to Python 3.12), it returns a 4-tuple.  Check for
+            # either here.
+            if result in (
+                (None, arg_string, None), (None, arg_string, None, None)
```

Comment:
Recommend a little "future proofing" here for clean line revisions.
```suggestion
                (None, arg_string, None),
                (None, arg_string, None, None),
```

Reply:
This got reverted by the latest push.